server/neptune/workflows/internal/revision_setter/workflow.go

```diff
@@ -0,0 +1,176 @@
+		StartToCloseTimeout: StartToCloseTimeout,
+		RetryPolicy: &temporal.RetryPolicy{
+			MaximumAttempts: GithubRetryCount,
+		},
+	})
+
+	var ga *activities.Github
+	var ra *activities.RevsionSetter
+	runner := &Runner{
+		GithubActivities:         ga,
+		RevisionSetterActivities: ra,
+		Scope:                    metrics.NewScope(ctx),
+	}
+
+	return runner.SetMiminumRevisionForRoot(ctx, request)
+}
+
+type Runner struct {
+	GithubActivities         githubActivities
+	RevisionSetterActivities revisionSetterActivities
+	Scope                    metrics.Scope
+}
+
+func (r *Runner) SetMiminumRevisionForRoot(ctx workflow.Context, request Request) error {
+	prs, err := r.listOpenPRs(ctx, request.Repo)
+	if err != nil {
+		return err
+	}
+
+	r.Scope.Counter("open_prs").Inc(int64(len(prs)))
+	if err := r.setMinRevisionForPRsModifiyingRoot(ctx, request, prs); err != nil {
+		return errors.Wrap(err, "setting minimum revision for pr modifiying root")
+	}
+	return nil
+}
+
+func (r *Runner) listOpenPRs(ctx workflow.Context, repo github.Repo) ([]github.PullRequest, error) {
+	var resp activities.ListPRsResponse
+	err := workflow.ExecuteActivity(ctx, r.GithubActivities.ListPRs, activities.ListPRsRequest{
+		Repo:  repo,
+		State: github.Open,
+	}).Get(ctx, &resp)
+	if err != nil {
+		return []github.PullRequest{}, errors.Wrap(err, "listing open PRs")
+	}
+
+	return resp.PullRequests, nil
+}
+
+func (r *Runner) setMinRevisionForPRsModifiyingRoot(ctx workflow.Context, req Request, prs []github.PullRequest) error {
+	// spawn activities to list modified files in each open PR async
+	futuresByPullNum := map[github.PullRequest]workflow.Future{}
+	for _, pr := range prs {
+		futuresByPullNum[pr] = workflow.ExecuteActivity(ctx, r.GithubActivities.ListModifiedFiles, activities.ListModifiedFilesRequest{
+			Repo:        req.Repo,
+			PullRequest: pr,
+		})
+	}
+
+	// resolve the futures and spawn activities to set minimum revision for PR if needed
+	futures := []workflow.Future{}
+	for _, pr := range prs {
+		if setMinimumRevisionFuture := r.setMinimumRevisionForPR(ctx, req, pr, futuresByPullNum[pr]); setMinimumRevisionFuture != nil {
+			futures = append(futures, workflow.ExecuteActivity(ctx, r.RevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+				Repository:  req.Repo,
+				PullRequest: pr,
+			}))
+		}
+	}
+
+	// wait to resolve futures for setting minimum revision
+	for _, future := range futures {
+		var resp activities.SetPRRevisionResponse
+		err := future.Get(ctx, &resp)
+		if err != nil {
+			return errors.Wrap(err, "error setting pr revision")
+		}
+	}
+
+	return nil
+}
+
+func (r *Runner) setMinimumRevisionForPR(ctx workflow.Context, req Request, pull github.PullRequest, future workflow.Future) workflow.Future {
+	setMinRevisionFn := func() workflow.Future {
+		return workflow.ExecuteActivity(ctx, r.RevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+			Repository:  req.Repo,
+			PullRequest: pull,
+		})
+	}
```

Comment:
Can you just extract this to a function, not sure there's a benefit to defining it within this function, just makes things harder to follow